File: app/src/main/java/com/example/ass07/customer/Room.kt
package com.example.ass07.customer


import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Room(
    @Expose
    @SerializedName("room_id") val room_id: Int,
    @Expose
    @SerializedName("room_type") val room_type :String,
    @Expose
    @SerializedName("price_per_day") val price_per_day:String,
    @Expose
    @SerializedName("status") val room_status:Int,
    @Expose
    @SerializedName("type_type_id") val emp_salary:Int,
    @Expose
    @SerializedName("name_type") val name_type:String,
    @Expose
    @SerializedName("type_id") val type_id:Int,
    @Expose
    @SerializedName("pet_type") val pet_type:Int,
    @Expose
    @SerializedName("image") val image:String,
    @Expose
    @SerializedName("user_id") val userId: Int
): Parcelable {}

File: app/data/routes/room.js
const express = require("express");
const router = express.Router();
const dbConn = require("../db");

// ดึงข้อมูลห้องพัก
router.get("/getroom", (req, res) => {
  let query = `
    SELECT r.room_id, rt.name_type AS room_type, rt.price_per_day,
    r.status AS room_status, pt.Pet_name_type AS pet_type
    FROM rooms r
    JOIN room_type rt ON r.type_type_id = rt.type_id
    JOIN pet_type pt ON rt.pet_type = pt.Pet_type_id
    WHERE r.deleted_at IS NULL
    AND rt.deleted_at IS NULL
    AND pt.deleted_at IS NULL;
  `;
  dbConn.query(query, (error, results) => {
    if (error) {
      return res.status(500).send({ error: true, message: "Internal Server Error" });
    }
    return res.json(results);
  });
});

// เพิ่มห้องพัก
router.post("/addroom", function (req, res) {
  const { type_type_id, status } = req.body;
  if (!type_type_id || status === undefined) {
    return res.status(400).send({ message: "Please provide room type ID and status" });
  }

  dbConn.query("SELECT name_type FROM room_type WHERE type_id = ?", [type_type_id], function (error, results) {
    if (error) {
      return res.status(500).send({ message: "Database error", details: error });
    }
    if (results.length === 0) {
      return res.status(400).send({ message: "Invalid room_type ID" });
    }
    const roomType = results[0];

    dbConn.query("INSERT INTO rooms (type_type_id, status) VALUES (?, ?)", [type_type_id, status], function (error, results) {
      if (error) {
        return res.status(500).send({ message: "Failed to insert room data", details: error });
      }
      return res.send({ message: "Room added successfully", id: results.insertId, room_type: roomType.name_type });
    });
  });
});

module.exports = router;
